tests/unit/data/data_pipeline/test_parquet_dataloader.py

```diff
@@ -0,0 +1,299 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the license found in the
+# LICENSE file in the root directory of this source tree.
+
+import os
+import random
+import shutil
+import string
+import tempfile
+import typing as tp
+import unittest
+from collections import Counter
+
+import numpy as np
+import numpy.typing as npt
+import pandas as pd
+import pyarrow as pa
+import pyarrow.parquet as pq
+
+from fairseq2.utils.parquet_dataloader import (
+    ParquetBasicDataLoader,
+    ParquetBasicDataloaderConfig,
+    ParquetBatchFormat,
+)
+
+
+def gen_random_string(length: int) -> str:
+    return "".join(
+        random.choice(string.ascii_letters + string.digits) for n in range(length)
+    )
+
+
+def generate_random_pandas_df(size: int, seed: int = 123) -> pd.DataFrame:
+    np_rs = np.random.RandomState(seed)
+    df: tp.Dict[str, tp.Union[npt.NDArray[tp.Any], tp.List[tp.Any]]] = {}
+    df["int_col"] = np_rs.randint(0, 200, size)
+    df["float_col"] = np_rs.randn(size)
+
+    df["string_col1"] = [gen_random_string(10) for _ in range(size)]
+    df["string_col2"] = [gen_random_string(2) for _ in range(size)]
+
+    df["list_int_col"] = [
+        np_rs.randint(-10, 10, np_rs.randint(0, 100)) for _ in range(size)
+    ]
+    df["list_float_col"] = [
+        np_rs.rand(np_rs.randint(0, 10)).astype(np.float32) for _ in range(size)
+    ]
+    df["list_float_fixed_size_col"] = [
+        np_rs.rand(7).astype(np.float32) for _ in range(size)
+    ]
+    return pd.DataFrame(df)
+
+
+def generated_partitioned_parquet_file(
+    path: str, size: int, n_partitions: int = 20, seed: int = 123
+) -> None:
+    df = generate_random_pandas_df(size, seed)
+
+    if n_partitions > 0:
+        df["part_key"] = np.arange(size) % n_partitions
+
+    table = pa.Table.from_pandas(df)
+
+    pq.write_to_dataset(
+        table,
+        path,
+        partition_cols=["part_key"] if n_partitions > 0 else None,
+        existing_data_behavior="delete_matching",
+        **{"row_group_size": 110},
+    )
+
+
+class TestParquetDataloader(unittest.TestCase):
```

Comment:
can you use pytest please ?
to get a temporary dir with pytest you just make a function with a `tmp_path: Path` as argument:

```py
def test_simple_dataload(tmp_path: Path):
        config = ParquetBasicDataloaderConfig(
            parquet_path=tmp_path / "test.pq", ...)
```